apps/aecore/src/aec_genesis_block_settings.erl

```diff
@@ -1,26 +1,46 @@
 -module(aec_genesis_block_settings).
 
 -export([dir/0,
-         preset_accounts/0]).
+         preset_accounts/0,
+         read_presets/0]).
 
 dir() ->
     filename:join(aeu_env:data_dir(aecore), ".genesis").
 
+-spec preset_accounts() -> list().
 preset_accounts() ->
-    PresetAccountsFile = filename:join([dir(), "accounts.json"]),
-    case file:read_file(PresetAccountsFile) of
-        {error, _Err} ->
+    case ?MODULE:read_presets() of
+        {error, _Err, PresetAccountsFile} ->
             % no setup, no preset accounts
             erlang:error({genesis_accounts_file_missing, PresetAccountsFile});
         {ok, JSONData} ->
+            DecodedData =
+                case JSONData of
+                    <<"{}">> ->
```

Comment:
This looks fragile e.g. for whitespaces/newlines before/in-middle/after the `{}`.